lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1842,11 +1855,21 @@ static void highlightOffendingType(TypeChecker &TC, InFlightDiagnostic &diag,
 static void checkGenericParamAccess(TypeChecker &TC,
                                     const GenericParamList *params,
                                     const Decl *owner,
+                                    bool checkUsableFromInline,
                                     AccessScope accessScope,
                                     AccessLevel contextAccess) {
   if (!params)
     return;
 
+  if (checkUsableFromInline) {
+    if (auto *VD = dyn_cast<ValueDecl>(owner)) {
+      if (VD->getFormalAccess() != AccessLevel::Internal)
```

Comment:
This is a suspicious condition—this could be above `internal` or it could be below `internal`.

Reply:
`checkUsableFromInline` means *only* check @usableFromInline and nothing else. So we call this function once with `checkUsableFromInline=false` if the decl is not @usableFromInline; if it is, we call it twice, first with `false` then with `true`.